CMakeLists.txt

```diff
@@ -615,149 +595,84 @@ if(QUDA_NVSHMEM)
-      GIT_TAG "${QIO_BRANCH}"
-      GIT_SHALLOW YES
-      PREFIX usqcd
-      CONFIGURE_COMMAND CC=${MPI_C_COMPILER} CXX=${MPI_CXX_COMPILER} <SOURCE_DIR>/configure "INSTALL=${INSTALL_EXE} -C"
-                        "CFLAGS=-Wall -O3 -std=c99 " --with-qmp=${QUDA_QMPHOME} --prefix=<INSTALL_DIR>
-                        --enable-largefile --disable-qmp-route --enable-dml-output-buffering
-      BUILD_COMMAND make
-      INSTALL_COMMAND make install
-      DEPENDS QMP
-      LOG_INSTALL ON
-      LOG_BUILD ON
-      LOG_DOWNLOAD ON)
-    ExternalProject_Get_Property(QIO INSTALL_DIR)
-    set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${INSTALL_DIR})
-    set(QUDA_QIOHOME
-        ${INSTALL_DIR}
-        CACHE PATH "path to QIO" FORCE)
-    set(QUDA_LIMEHOME
-        ${INSTALL_DIR}
-        CACHE PATH "path to LIME" FORCE)
-    ExternalProject_Add_Step(
-      QIO reconf
-      COMMAND ${AUTORECONF_EXE} -fi
-      WORKING_DIRECTORY <SOURCE_DIR>
-      DEPENDERS configure
-      DEPENDEES download)
-    set(QUDA_QIO_LDFLAGS
-        "-L${QUDA_QIOHOME}/lib"
-        CACHE STRING "LDFLAGS for QMP - should be derived from qmp-config --ldflags")
-    set(QUDA_QIO_LIBS
-        "-lqio -llime"
-        CACHE STRING "LIBS for QMP - should be derived from qmp-config --libs")
-
-  else()
-    if("${QUDA_QIOHOME}" STREQUAL "" OR "${QUDA_LIMEHOME}" STREQUAL "")
-      message(
-        FATAL_ERROR "QUDA_QIOHOME and QUDA_LIMEHOME must be defined when QUDA_QIO is ON and QUDA_DOWNLOAD_USQCD is OFF")
-    endif()
-    execute_process(COMMAND ${QUDA_QIOHOME}/bin/qio-config --cflags OUTPUT_VARIABLE QUDA_QIO_CFLAGS
-                                                                                    OUTPUT_STRIP_TRAILING_WHITESPACE)
-    execute_process(COMMAND ${QUDA_QIOHOME}/bin/qio-config --ldflags OUTPUT_VARIABLE QUDA_QIO_LDFLAGS_INTERNAL
-                                                                                     OUTPUT_STRIP_TRAILING_WHITESPACE)
-    execute_process(COMMAND ${QUDA_QIOHOME}/bin/qio-config --libs OUTPUT_VARIABLE QUDA_QIO_LIBS_INTERNAL
-                                                                                  OUTPUT_STRIP_TRAILING_WHITESPACE)
-    set(QUDA_QIO_LDFLAGS
-        ${QUDA_QIO_LDFLAGS_INTERNAL}
-        CACHE STRING "LDFLAGS for QMP - should be derived from qmp-config --ldflags")
-    set(QUDA_QIO_LIBS
-        ${QUDA_QIO_LIBS_INTERNAL}
-        CACHE STRING "LIBS for QMP - should be derived from qmp-config --libs")
+    CPMAddPackage(
+      NAME QIO
+      GITHUB_REPOSITORY usqcd-software/qio
+      GIT_TAG 3c99383
+      OPTIONS 
+        "QIO_ENABLE_PARALLEL_BUILD ON"
+        "QIO_ENABLE_QMP_ROUTE OFF" 
+        "QIO_ENABLE_OUTPUT_BUFFERING ON" 
+        "QIO_DML_BUF_BYTES 33554432"
+    )
   endif()
 endif()
 
-if(QUDA_OPENMP)
-  find_package(OpenMP)
+####################################
+# USQCD QDPJIT
+####################################
+if(QUDA_QDPJIT)
```

Comment:
I CMakeified QDP_JIT so this will get simpler soon. I just need to see if it still attempts to hijack your allocator or not. It should transitively bring QMP, QIO etc.

Reply:
Cool. Happy to see that also getting cmakified.